broker/lib/fabrik/DirectorInstance.js

```diff
@@ -240,21 +243,60 @@ class DirectorInstance extends BaseInstance {
         );
     }
     logger.info('Fetching state of last operation', operation);
-    return Promise
-      .try(() => {
-        assert.ok(operation.task_id, 'Operation must have the property \'task_id\'');
-        return this.manager.getTask(operation.task_id);
-      })
-      .catchThrow(NotFound, new ServiceInstanceNotFound(this.guid))
-      .then(task => {
-        assert.ok(_.endsWith(task.deployment, this.guid), `Deployment '${task.deployment}' must end with '${this.guid}'`);
-        this.networkSegmentIndex = this.manager.getNetworkSegmentIndex(task.deployment);
-        this.setOperationState(operation, task);
-        if (operation.state !== 'in progress') {
-          return this.finalize(operation);
+    const instanceId = this.guid;
+
+    const getBoshTaskOperation = (operation, taskId, removeFromCache) => {
+      return Promise
+        .try(() => {
+          assert.ok(taskId, 'Task ID must be available');
+          return this.manager.getTask(taskId);
+        })
+        .catchThrow(NotFound, new ServiceInstanceNotFound(instanceId))
+        .then(task => {
+          assert.ok(_.endsWith(task.deployment, this.guid), `Deployment '${task.deployment}' must end with '${this.guid}'`);
+          this.networkSegmentIndex = this.manager.getNetworkSegmentIndex(task.deployment);
+          this.setOperationState(operation, task);
+          if (operation.state !== 'in progress') {
+            return Promise.try(() => {
+                if (removeFromCache) {
+                  return utils
+                    .retry(tries => {
+                      logger.info(`+-> ${ordinals[tries]} attempt to delete cached bosh task for '${instanceId}'...`);
+                      return this.manager.removeCachedTask(instanceId);
+                    }, {
+                      maxAttempts: 5,
+                      minDelay: 1000
+                    })
+                    .catch(err => {
+                      logger.error('Error removing task from ETCD Cache after multiple attempts', err);
+                      throw err;
+                    });
+                }
+              })
+              .then(() => this.finalize(operation));
+          }
+        })
+        .return(operation);
+    };
+
+    if (operation.task_id) {
+      return getBoshTaskOperation(operation, operation.task_id, false);
+    } else {
+      return Promise.try(() => {
+        return this.manager.getCurrentOperationState(this.guid);
+      }).then(state => {
+        const isCached = state.cached;
```

Comment:
I actually thought of naming this as `queued` but named this `cached` instead to avoid a naming confusion with the bosh task's `queued` state. So, `cached` should indicate if the operation is staggered and available in etcd.

Reply:
ok.